### PR TITLE
Feature/should add internationalization

### PR DESCRIPTION
**Pull request by L1ght-time**

### CODE REVIEW

File: src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.constants.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 export const modelLanguages = [
-  { id: 1, name: 'Ukrainian' },
+  { id: 1, name: 'Ukraine' },
   { id: 2, name: 'English' },
 ];
```
Comment (mykyta-lazurenko) [inline src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.constants.ts:4]:
```suggestion
] as const;
```

File: src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.tsx [hunk 2]
```diff
@@ -9,11 +9,17 @@ import {
...
 } from '@/components/shared/DropDown';
+import { LOCALES } from '@/i18n';
 
 import { modelLanguages } from './LanguageDropdown.constants';
 
 export const LanguageDropdown = () => {
-  const { language, setLanguage } = useLayoutStore();
+  const { language, setLocale, setLanguage } = useLayoutStore();
+
+  const handleClick = (name: string) => {
```
Comment (mykyta-lazurenko) [inline src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.tsx:19]: `useCallback`

File: src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.tsx [hunk 3]
```diff
@@ -9,11 +9,17 @@ import {
...
+    setLanguage(name);
+    setLocale(LOCALES[name.toUpperCase() as keyof typeof LOCALES]);
```
Comment (mykyta-lazurenko) [inline src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.tsx:21]: Move this logic in `useLayoutStore()`, when changing locales we need both functions. Could you combine `setLanguage` and `setLocale`?

Comment (mykyta-lazurenko) [inline src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.tsx:21]: `keyof typeof LOCALES` move to type `LocalesKeys` near locales and import here.

File: src/components/Layout/NavBar/LanguageDropdown/constants.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const LANGUAGES = {
+  ENGLISH: 'ENGLISH',
+  UKRAINE: 'UKRAINE',
+};
```
Comment (mykyta-lazurenko) [inline src/components/Layout/NavBar/LanguageDropdown/constants.ts:4]: Better use camelCase for keys

Comment (mykyta-lazurenko) [inline src/components/Layout/NavBar/LanguageDropdown/constants.ts:4]:
```suggestion
export const LANGUAGES = {
  ENGLISH: 'ENGLISH',
  UKRAINE: 'UKRAINE',
} as const;
```

File: src/i18n/I18nProvider/I18nProvider.types.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { PropsWithChildren } from 'react';
+
+import { LOCALES } from '@/i18n';
+
+export type TProvider = PropsWithChildren & { locale: keyof typeof LOCALES };
```
Comment (mykyta-lazurenko) [inline src/i18n/I18nProvider/I18nProvider.types.ts:5]: reuse `keyof typeof LOCALES` after creating type `LocalesKeys`

Comment (mykyta-lazurenko) [reply src/i18n/I18nProvider/I18nProvider.types.ts:5]: It's props for `IntlProvider`, inherit him props.

File: src/i18n/I18nProvider/I18nProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Fragment } from 'react';
+import { IntlProvider } from 'react-intl';
+
+import messages from '../messages';
+
+import { TProvider } from './I18nProvider.types';
+
+export const I18nProvider = ({ children, locale }: TProvider) => (
+  <IntlProvider textComponent={Fragment} locale={locale} messages={messages[locale]}>
```
Comment (mykyta-lazurenko) [inline src/i18n/I18nProvider/I18nProvider.tsx:9]: After update `TProvider` type you can use rest and spread operators like that:

```suggestion
export const I18nProvider = ({ children, locale, ...props }: TProvider) => (
  <IntlProvider textComponent={Fragment} locale={locale} messages={messages[locale]} {...props} >
```

File: src/store/layoutStore.ts [hunk 7]
```diff
@@ -1,18 +1,28 @@
...
   setOpen: () => void;
   language: string;
   setLanguage: (value: string) => void;
+  locale: string;
+  setLocale: (value: string) => void;
 }
 
 export const useLayoutStore = create<IStore>((set) => ({
   isOpen: true,
   setOpen: () => set((state: IStore) => ({ isOpen: !state.isOpen })),
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:17]: It's toggle open, better rename to `toggleOpen`

File: src/i18n/constants.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export const LOCALES = {
+  ENGLISH: 'en-us',
+  UKRAINE: 'uk-ua',
+};
```
Comment (mykyta-lazurenko) [inline src/i18n/constants.ts:4]: Better use camelCase for keys.

Comment (mykyta-lazurenko) [reply src/i18n/constants.ts:4]: In most cases, locale keys save as 2 letters, like `en`, `ru`, `ua`, `it`.

File: src/pages/_app.tsx [hunk 9]
```diff
@@ -13,10 +16,14 @@ type AppPropsWithLayout = AppProps & {
   Component: NextPageWithLayout;
 };
 function MyApp({ Component, pageProps }: AppPropsWithLayout) {
+  const { locale } = useLayoutStore();
+
   return (
-    <Layout>
-      <Component {...pageProps} />
-    </Layout>
+    <I18nProvider locale={locale as keyof typeof LOCALES}>
```
Comment (mykyta-lazurenko) [inline src/pages/_app.tsx:22]: Remove `as` operator

File: src/store/layoutStore.ts [hunk 10]
```diff
@@ -1,18 +1,28 @@
...
 
+import { LOCALES } from '@/i18n';
+import { LANGUAGES } from '@/components/Layout/NavBar/LanguageDropdown/constants';
+
 interface IStore {
   isOpen: boolean;
   setOpen: () => void;
   language: string;
   setLanguage: (value: string) => void;
+  locale: string;
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:11]: It's not `string`, it's a literals union

File: src/store/layoutStore.ts [hunk 11]
```diff
@@ -1,18 +1,28 @@
...
+  setLocale: (value: string) => void;
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:12]: `value` isn't `string`, it's a literals union

File: src/store/layoutStore.ts [hunk 12]
```diff
@@ -1,18 +1,28 @@
...
   setOpen: () => set((state: IStore) => ({ isOpen: !state.isOpen })),
-  language: 'English',
+  language: LANGUAGES.ENGLISH,
   setLanguage: (language) =>
     set((state: IStore) => ({
       language,
     })),
+  locale: LOCALES.ENGLISH,
+  setLocale: (locale) =>
+    set((state: IStore) => ({
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:25]: You do not use `state`, remove him.
```suggestion
    set(() => ({
```

File: src/store/layoutStore.ts [hunk 13]
```diff
@@ -1,18 +1,28 @@
...
+  setLocale: (value: string) => void;
 }
 
 export const useLayoutStore = create<IStore>((set) => ({
   isOpen: true,
   setOpen: () => set((state: IStore) => ({ isOpen: !state.isOpen })),
-  language: 'English',
+  language: LANGUAGES.ENGLISH,
   setLanguage: (language) =>
     set((state: IStore) => ({
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:20]: You do not use `state`, remove him.

File: src/store/layoutStore.ts [hunk 14]
```diff
@@ -1,18 +1,28 @@
...
       language,
     })),
+  locale: LOCALES.ENGLISH,
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:23]: What is the difference between the keys `locale` and `language`?

File: src/i18n/helpers/translate.tsx [hunk 15]
```diff
@@ -0,0 +1,3 @@
+import { FormattedMessage } from 'react-intl';
+
+export const translate = (id: string, value = {}) => <FormattedMessage id={id} values={{ ...value }} />;
```
Comment (mykyta-lazurenko) [inline src/i18n/helpers/translate.tsx:3]: It's a component, not a helper, he returns `jsx`.

Comment (mykyta-lazurenko) [inline src/i18n/helpers/translate.tsx:3]: You can use `formattedMessage` from `react-intl` instead.

**Review by mykyta-lazurenko — CHANGES_REQUESTED** [review]

File: src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.tsx (hunk 2 above)
Comment (L1ght-time) [reply src/components/Layout/NavBar/LanguageDropdown/LanguageDropdown.tsx]: I don't use handleClick anymore.

File: src/store/layoutStore.ts [hunk 16]
```diff
@@ -1,18 +1,20 @@
...
 
+import { LOCALES, TLocalesKeys } from '@/i18n';
+
 interface IStore {
-  isOpen: boolean;
-  setOpen: () => void;
-  language: string;
-  setLanguage: (value: string) => void;
+  toggleOpen: boolean;
+  setToggleOpen: () => void;
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:7]:
```suggestion
toggleOpen: () => void;
```

File: src/store/layoutStore.ts [hunk 17]
```diff
@@ -1,18 +1,20 @@
 import { create } from 'zustand';
 
+import { LOCALES, TLocalesKeys } from '@/i18n';
+
 interface IStore {
-  isOpen: boolean;
-  setOpen: () => void;
-  language: string;
-  setLanguage: (value: string) => void;
+  toggleOpen: boolean;
```
Comment (mykyta-lazurenko) [inline src/store/layoutStore.ts:6]:
```suggestion
  isOpen: boolean;
```

File: src/pages/_app.tsx [hunk 18]
```diff
@@ -13,10 +15,14 @@ type AppPropsWithLayout = AppProps & {
   Component: NextPageWithLayout;
 };
 function MyApp({ Component, pageProps }: AppPropsWithLayout) {
+  const { locale } = useLayoutStore();
```
Comment (mykyta-lazurenko) [inline src/pages/_app.tsx:18]: Move to `I18nProvider`

**Review by mykyta-lazurenko — CHANGES_REQUESTED** [review]

**Review by mykyta-lazurenko — APPROVED** [review]

**Pull request merged**